internal/service/ec2/ec2_instance.go

```diff
@@ -2635,16 +2640,20 @@ func buildInstanceOpts(d *schema.ResourceData, meta interface{}) (*awsInstanceOp
 		AvailabilityZone: aws.String(d.Get("availability_zone").(string)),
 	}
 
-	if v := d.Get("placement_group").(string); instanceInterruptionBehavior == "" || instanceInterruptionBehavior == ec2.InstanceInterruptionBehaviorTerminate {
-		opts.Placement.GroupName = aws.String(v)
-		opts.SpotPlacement.GroupName = aws.String(v)
+	if v, ok := d.GetOk("placement_group"); ok && (instanceInterruptionBehavior == "" || instanceInterruptionBehavior == ec2.InstanceInterruptionBehaviorTerminate) {
```

Comment:
This is the crux of the problem noted by op. `ec2.RunInstanceInput.Placement.GroupName` cannot be set if launching in a host resource group. This can be done implicitly based on the AMI ID and the host resource group being set to "Allocate hosts automatically." `placement_group` would always be set, even if to an empty string, if instance interruption behavior was not set or set to terminate.